tests/TestCase/Middleware/ThrottleMiddlewareTest.php

```diff
@@ -0,0 +1,280 @@
+<?php
+namespace Muffin\Throttle\Test\TestCase\Middleware;
+
+use Cake\Cache\Cache;
+use Cake\Http\ServerRequest;
+use Cake\Http\Response;
+use Cake\TestSuite\TestCase;
+use Muffin\Throttle\Middleware\ThrottleMiddleware;
+use StdClass;
+
+class ThrottleMiddlewareTest extends TestCase
+{
+    /**
+     * setUp method
+     *
+     * @return void
+     */
+    public function setUp()
+    {
+        parent::setUp();
+
+        $this->skipIf(!function_exists('apcu_store'), 'APCu is not installed or configured properly.');
```

Comment:
Also skip the tests for CakePHP < 3.3 as middleware support is not available before that. I want to add this middleware without bumping the plugin's CakePHP version dependency.